doc/install-on-docker.md

```diff
@@ -23,21 +66,45 @@ cd ray
 
 This script creates several Docker images:
 
- * The `ray-project/ray:deploy` image is a self-contained copy of code and binaries suitable for end users.
- * The `ray-project/ray:examples` adds additional libraries for running examples.
- * Ray developers who want to edit locally on the host filesystem should use the `ray-project/ray:devel` image, which allows local changes to be reflected immediately within the container.
+ * The `ray-project/deploy` image is a self-contained copy of code and binaries suitable for end users.
+ * The `ray-project/examples` adds additional libraries for running examples.
+ * The `ray-project/base-deps` image builds from Ubuntu Xenial and includes Anaconda and other basic dependencies and can serve as a starting point for developers.
+
+Review images by listing them:
+```
+$ docker images
+```
+
+Output should look something like the following:
+```
+REPOSITORY                          TAG                 IMAGE ID            CREATED             SIZE
+ray-project/examples                latest              7584bde65894        4 days ago          3.257 GB
+ray-project/deploy                  latest              970966166c71        4 days ago          2.899 GB
+ray-project/base-deps               latest              f45d66963151        4 days ago          2.649 GB
+ubuntu                              xenial              f49eec89601e        3 weeks ago         129.5 MB
+```
+
 
 ## Launch Ray in Docker
 
 Start out by launching the deployment container.
 
 ```
-docker run --shm-size=1024m -t -i ray-project/ray:deploy
+docker run --shm-size=<shm-size> -t -i ray-project/ray:deploy
+```
+
+Replace `<shm-size>` with a limit appropriate for your system, for example `512M` or `2G`.
+The `-t` and `-i` options here are required to support interactive use of the container.
+You should now see a prompt that looks something like:
+
+```
+root@ebc78f68d100:/ray#
 ```
 
+
 ## Test if the installation succeeded
 
-To test if the installation was successful, try running some tests.
+To test if the installation was successful, try running some tests. Within the container shell enter the following commands:
 
 ```
 python test/runtest.py # This tests basic functionality.
```

Comment:
I just tried this and it told me 

```
Traceback (most recent call last):
  File "test/runtest.py", line 127, in <module>
    raise Exception("You have an older version of cloudpickle that is not able to serialize namedtuples. Try running \n\n{}\n\n".format(cloudpickle_command))
Exception: You have an older version of cloudpickle that is not able to serialize namedtuples. Try running 

pip install --upgrade cloudpickle
```

We should make sure that we're installing the correct version of cloudpickle in the docker image. For that it should be sufficient to do `pip install cloudpickle` somewhere.